### DISCUSSION TITLE
Regarding the release of the OrchardCore.Cms.Web project

### DISCUSSION THREAD

**Discussion by fuxiaojun0203** [OrchardCMS/OrchardCore#15841] [Q&A]
After publishing the OrchardCore.Cms.Web project, I want to keep the front-end code files, such as the directories in the theme file, so that front-end engineers can use the editor such as vscode to modify the web page, instead of Vs2019, how do I set up?
Or, after the release, there are only .dll files and this cannot be changed.

**Comment by jptissot** [comment]
When you say "front-end code files" what do you mean ? Do you mean the liquid templates ? Or the javascript / css files ?

**Comment by fuxiaojun0203** [comment]
the liquid templates and javascript / css files 

![QQ截图20200512084834](https://user-images.githubusercontent.com/25946521/81626124-62c8fe00-942d-11ea-88e6-21f98a601295.png)

**Comment by jptissot** [comment]
For the liquid templates, you could potentially use the `Design ->Templates` section in the admin panel. 
I have never tried this, but you could also write a [TemplatesShapeBindingResolver ](https://github.com/OrchardCMS/OrchardCore/blob/ccb86d6cd49c5ae504bbabce7c6b5144f9916c75/src/OrchardCore.Modules/OrchardCore.Templates/Services/TemplatesShapeBindingResolver.cs#L13) that would look on the file system for liquid templates? 

For the CSS/JS files, you could allow these types of files in the media library and allow your designers to upload them there (probably a security risk). You could also write your own [IModuleStaticFileProvider](https://github.com/OrchardCMS/OrchardCore/blob/6cddd9f9904ff117ba82587f39fc76f49fac3cc3/src/OrchardCore/OrchardCore.Abstractions/Modules/ModuleEmbeddedStaticFileProvider.cs#L13)
That looks on the file system for assets.

**Comment by ns8482e** [comment]
@fuxiaojun0203 I am using VSCode for development without any issue.

Once the project published all your files embedded in published .net assembly, so you won’t be able to change after publish. I think the best approach to use with CI/CD

If you would like to change views after publish you need to use Template module as suggested by @jptissot

**Comment by fuxiaojun0203** [comment]
@ns8482e   After the project is released, can't I keep the view files and JavaScript / css files in the theme?    Instead of only TheMenksoftTheme.dll file

And also，What does CI / CD mean?

**Comment by jptissot** [comment]
CI / CD is continuous integration and delivery. It usually means that any code changes will trigger a build and push the updated dll to your site.

This is a better avenue because you get the benefits of using Git to track the history of the templates and static assets.
